storage/src/main/scala/coop/rchain/storage/regex/PathRegex.scala

```diff
@@ -65,10 +120,77 @@ private[regex] object PathToken {
 case class PathRegex(tokens: List[PathToken], options: RegexOptions) {
   def withOptions(newOptions: RegexOptions): PathRegex = new PathRegex(tokens, newOptions)
 
-  def toPath(args: List[(String, Any)] = Nil): Either[Throwable, String] =
-    throw new NotImplementedError("TODO")
+  /**
+    * Takes some arguments (argument can be a value or a sequence of values),
+    * and builds an Uri-path
+    */
+  def toPath(args: Map[String, Iterable[String]],
+             encode: String => String = PathRegex.encodeUriComponent): Either[Throwable, String] =
+    Try(tokens.map(_.formatSegment(args, encode)).mkString).toEither
 
-  def toPath(arg: (String, Any)): Either[Throwable, String] = toPath(arg :: Nil)
+  val keys: List[PathToken] = tokens.filter(_.isToken)
+
+  /**
+    * Returns Regex that is able to parse our tokens with defined options
+    */
+  lazy val Regex: Either[Throwable, Regex] = {
```

Comment:
A couple questions:

1. Why the `lazy val` instead of a `def`?
2. Why capitalize `Regex`?  It's a bit confusing when we also have a class named `Regex`

Reply:
1. The PathRegex has two different functions: a) parse existing paths (and then Regex object is required) b) Create path from parsed string and args provided (and then Regex object is unused); I've used lazy val assuming that someone will save an instance of PathRegex object in scenario (a) and recreation of that object with every call will be a waste of resources.

2. I just named it wrong :( - at prev. work we have camelCase for private members/variables/args and PascalCase for public members, need some time to adopt to new style everywhere. Will be more careful in future.

Reply:
1. Seems reasonable
2. No problem.  Occasionally you see that kind of naming approach in Scala when you want a function call too look and smell like a case class constructor, but it's kinda rare.  Lower camel-case is definitely the convention.